src/main/scala/it/pagopa/interop/partyprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -393,22 +394,44 @@ class ProcessApiServiceImpl(
         }
       }
       contractTemplate   <- getFileAsString(onboardingRequest.contract.path)
-      pdf                <- pdfCreator.createContract(contractTemplate, validManager, validUsers, institution)
-      digest             <- signatureService.createDigest(pdf)
-      token              <- partyManagementService.createToken(
+      pdf    <- pdfCreator.createContract(contractTemplate, validManager, validUsers, institution, onboardingRequest)
+      digest <- signatureService.createDigest(pdf)
+      token  <- partyManagementService.createToken(
         PartyManagementDependency.Relationships(relationships),
         digest,
         onboardingRequest.contract.version,
         onboardingRequest.contract.path
       )(bearer)
       _ = logger.info("Digest {}", digest)
       onboardingMailParameters <- getOnboardingMailParameters(token.token, currentUser, onboardingRequest)
-      destinationMails = ApplicationConfiguration.destinationMails.getOrElse(Seq(institution.digitalAddress))
+      destinationMails = ApplicationConfiguration.destinationMails.getOrElse(
+        Seq(onboardingRequest.institutionUpdate.flatMap(_.digitalAddress).getOrElse(institution.digitalAddress))
+      )
       _ <- sendOnboardingMail(destinationMails, pdf, onboardingMailParameters)
       _ = logger.info(s"$token")
     } yield ()
   }
 
+  private def validateOverridingData(
+    onboardingRequest: OnboardingSignedRequest,
+    institution: PartyManagementDependency.Institution
+  ): Future[Unit] = {
+    val success = if (institution.origin === "IPA") {
+      onboardingRequest.institutionUpdate.forall(updates =>
+        updates.description.forall(_.equals(institution.description)) &&
+          updates.taxCode.forall(_.equals(institution.taxCode)) &&
+          updates.digitalAddress.forall(_.equals(institution.digitalAddress)) &&
+          updates.zipCode.forall(_.equals(institution.zipCode)) &&
+          updates.address.forall(_.equals(institution.address))
+      )
+    } else {
+      true
+    }
+
+    if (success) Future.successful(())
+    else Future.failed(OnboardingInvalidUpdates(institution.externalId))
+  }
```

Comment:
Not sure, but I think the version belove should obtain the same result:
```
  private def validateOverridingData(
    onboardingRequest: OnboardingSignedRequest,
    institution: PartyManagementDependency.Institution
  ): Future[Unit] = {
    val areIpaDataMatching: Boolean = onboardingRequest.institutionUpdate.forall(updates =>
      updates.description.contains(institution.description) ||
        updates.taxCode.contains(institution.taxCode) ||
        updates.digitalAddress.contains(institution.digitalAddress) ||
        updates.zipCode.contains(institution.zipCode) ||
        updates.address.contains(institution.address)
    )

    Future
      .failed(OnboardingInvalidUpdates(institution.externalId))
      .unlessA(institution.origin != "IPA" || areIpaDataMatching)
  }
```
In Scala, you can use directly `==` instead of `.equals` or choose to use `Eq typeclass` from cats lib to obtain more strict equality.
`.forall` function, by definition, returns `true` if the object is empty (None in Option context), so in this case, you can achieve the same result using `contains` function and `||` operator.

Reply:
@beetlecrunch 
This way you are changing the business logic:

1. when origin is IPA
1.1. if at least one of the input fields differs from persisted value, we have to give an error
1.2. A field not overridden (thus not in input) it's considered to be ok

"contains" function return false if not present, this is invalidating point 1.2
The OR condition invalidates the positive logic expressed by the areIpaDataMatching: if just one of the condition is true, all the expression is considered true: even if there is at least one field overridden

ps: my unitTests are not failing as I was supposing, maybe I'm misunderstanding your solution :)

Reply:
very minor addition to the suggested change: `areIpaDataMatching` could be declared `lazy` like
`lazy val isIpaDataMatching: Boolean = ...`
this way checks are not even executed if the origin is different from IPA

Reply:
Ok, I got your point, but actually, the test works also with my solution. Probably it must be enforced.
May I suggest you use this function?
```
  private def validateOverridingData(
    onboardingRequest: OnboardingSignedRequest,
    institution: PartyManagementDependency.Institution
  ): Future[Unit] = {
    val areIpaDataMatching: Boolean = onboardingRequest.institutionUpdate.forall(updates =>
      updates.description.forall(_ == institution.description) &&
        updates.taxCode.forall(_ == institution.taxCode) &&
        updates.digitalAddress.forall(_ == institution.digitalAddress) &&
        updates.zipCode.forall(_ == institution.zipCode) &&
        updates.address.forall(_ == institution.address)
    )

    Future
      .failed(OnboardingInvalidUpdates(institution.externalId))
      .unlessA(institution.origin != "IPA" || areIpaDataMatching)
  }
```

Reply:
@beetlecrunch @galales do you know how could I correctly check if the mock is never called?

I'm pushing the code with the error that should brought to the failuer with a TODO to remember to fix it

Reply:
This test does not work as expected due to the too generic use of 400 status error.
I change the error handler and now it fails correctly.
I leave the test with the wrong version, so you can verify the behavior by yourself.

Reply:
I see the suggested fix and decided to return 409 when the conflict occur :)